play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -585,6 +586,101 @@ class AvatarsPanel extends PSRoomPanel {
+	static readonly location = 'semimodal-popup';
+	static readonly noURL = true;
+
+	handleForfeit = (ev: Event) => {
+		const elem = this.props.room.parentElem;
+		const roomid = (elem as HTMLInputElement)?.value as RoomID || PS.getRoom(elem)?.id || '' as RoomID;
+		const room = PS.rooms[roomid] as BattleRoom;
+
+		const closeAfter = this.base!.querySelector<HTMLInputElement>('input[name=closeroom]')?.checked;
+		room.send("/forfeit");
+		if (closeAfter) PS.leave(room.id);
+		ev.preventDefault();
+		this.close();
+	};
+
+	update = () => {
+		this.forceUpdate();
+	};
+
+	override render() {
+		const room = this.props.room;
+		const elem = room.parentElem;
+		const roomid = (elem as HTMLInputElement)?.value as RoomID || PS.getRoom(elem)?.id || '' as RoomID;
+		const battleRoom = PS.rooms[roomid] as BattleRoom;
+
+		return <PSPanelWrapper room={room} width={480}><div class="pad">
+			<form>
+				<p>Forfeiting makes you lose the battle. Are you sure?</p>
+				<p>
+					<label class="checkbox"><input type="checkbox" name="closeroom" checked={true}></input> Close after
+						forfeiting</label>
+				</p>
+				<p>
+					<button onClick={this.handleForfeit} class="button"><strong>Forfeit</strong></button> {}
+					{!battleRoom.battle.rated && <button type="button" value={battleRoom.id} data-href="replaceplayer" class="button">
+						Replace player
+					</button>} {}
+					<button type="button" name="close" data-cmd="/close" class="button autofocus">
+						Cancel
+					</button>
+				</p>
+			</form>
+
+		</div></PSPanelWrapper>;
+	}
+}
+
+class ReplacePlayerPanel extends PSRoomPanel {
+	static readonly id = 'replaceplayer';
+	static readonly routes = ['replaceplayer'];
+	static readonly location = 'semimodal-popup';
+
+	handleReplacePlayer = (ev: Event) => {
+		const elem = this.props.room.parentElem;
+		const roomid = (elem as HTMLInputElement)?.value as RoomID || PS.getRoom(elem)?.id || '' as RoomID;
+		const room = PS.rooms[roomid] as BattleRoom;
+		const newPlayer = this.base?.querySelector<HTMLInputElement>("input[name=newplayer]")?.value;
+		if (!newPlayer?.length) return room.add("|error|Enter player's name");
+		if (room.battle.ended) return room.add("|error|Cannot replace player, battle has already ended.");
+		let playerSlot = room.battle.p1.id === PS.user.userid ? "p1" : "p2";
+		room.send('/leavebattle');
+		room.send(`/addplayer ${newPlayer}, ${playerSlot}`);
+		this.close();
+		ev.preventDefault();
+	};
+
+	update = () => {
+		this.forceUpdate();
+	};
+
+	override render() {
+		const room = this.props.room;
+
+		return <PSPanelWrapper room={room} width={480}><div class="pad">
+			<form onSubmit={this.handleReplacePlayer}>
+				<p>Replacement player's name:</p>
+				<p>
+					<input name="newplayer" class="textbox autofocus" />
+				</p>
+				<p>
+					<button type="submit" class="button"><strong>Replace</strong></button>
+
+					<button type="button" data-cmd="/close" class="button autofocus">
```

Comment:
```suggestion
					<button type="button" data-cmd="/close" class="button">
```
Should obviously not be the autofocus.